xmodule/library_content_block.py

```diff
@@ -714,10 +714,7 @@ def definition_from_xml(cls, xml_object, system):
                 if system.error_tracker is not None:
                     system.error_tracker(msg)
 
-        definition = {
-            attr_name: json.loads(attr_value)
-            for attr_name, attr_value in xml_object.attrib.items()
-        }
+        definition = dict(xml_object.attrib.items())
```

Comment:
@yusuf-musleh What's the explanation for this error, and do we know this fix won't cause other issues?

Reply:
In particular, does this affect the import/export of a course that uses library content blocks.

Reply:
@bradenmacdonald  Alright, so I took a deeper look. Like I mentioned in my previous comment, the issue was caused because we were trying to load JSON from values that were not JSON. Here are some examples XMLs for the LibraryContentBlocks with their respective `xml_object.attrib.items()` values:

**Normal Course with LibraryContentBlock Unit**
```xml
<library_content source_library_id="library-v1:DemoX+DL1" source_library_version="64f4927af269d40f9daf89f8" x-is-pointer-node="no" url_name="library_content11">

<!-- some content here -->

</library_content>
```
The value of `xml_object_attrib.items()`

```python
[('x-is-pointer-node', 'no'), ('url_name', 'library_content11')]
```

**Imported Course with LibraryContentBlock Unit**

```xml
<library_content source_library_id="library-v1:DemoX+DL1" source_library_version="64f4927af269d40f9daf89f8" copied_from_block="block-v1:edX+DemoX3+1+type@library_content+block@fb56a2f2b76f4a84807af4eff2749342" x-is-pointer-node="no" url_name="library_content19">

<!-- some content here -->

</library_content>
```
The value of `xml_object_attrib.items()`
```python
[('copied_from_block', 'block-v1:edX+DemoX3+1+type@library_content+block@fb56a2f2b76f4a84807af4eff2749342'), ('x-is-pointer-node', 'no'), ('url_name', 'library_content19')]
```

So the `json.loads` on the values wouldn't work here, which was the main issue, and once that was removed, the pylint suggested to use `dict` since there were no changes done to the values.

> In particular, does this affect the import/export of a course that uses library content blocks.

I tested exporting/importing a course that contains the LibraryContentBlocks, and it seems to be working fine, so no issues there.

Just to be safe, since there might be some reasoning behind having the `json.loads` there that I am not aware of, would it be better to just add a try/except and handle it like so:

```python
        definition = {}
        for attr_name, attr_value in xml_object.attrib.items():
            try:
                definition[attr_name] = json.loads(attr_value)
            except json.JSONDecodeError:
                definition[attr_name] = attr_value
```

Reply:
No, I don't think we need the try/except. I'd just like to understand why the `json.loads` was there before, and why it's not needed now - what changed? Why wasn't this same bug happening when importing a library content block in a course, before your fix?

The change could have been some of the other work that @Agrendalath and I have been doing on the runtime though. I doubt it's anything specific to copy-paste.

Reply:
Ah, now I see - it's because [`clean_metadata_from_xml`](https://github.com/openedx/edx-platform/blob/548d302610b9f39edd457c4c50b5feba38f2b188/xmodule/xml_block.py#L161-L171) is stripping out the `Scope.settings` fields before this function is called. The prior version generally wasn't seeing any fields at all, so it didn't matter whether it was calling JSON.loads or not.